### PR TITLE
Document Installing Antora's dependencies & Updating the Search Index

### LINKED ISSUE #416: Readme is missing the command line preparation to run the antora command
## What Type Of Content Change Is This?

- [ ] New Content Addition
- [ ] Old Content Deprecation
- [ ] Existing Content Simplification
- [x] Bug Fix to Existing Content

## Which Manual Does This Relate To?

- [ ] Admin Manual
- [ ] Developer Manual
- [ ] User Manual
- [x] General

## What Needs to be Documented?

In the current docs/readme, there is, at least I did not find it, no guide to install/update the command line environment for antora. We have package.json, which tells what to install, but the command 
(I think it is `npm i`) is missing as prerequisite / update.
Note: it should also be noted to run `npm i` on a regular basis to update changes made in the prerequisites 

## Why Should This Change Be Made?

Fix missing antora command line preperation

### PR DESCRIPTION
**Pull request by settermjd**
This fixes #416. It adds a section to the top of docs/build-the-docs.md that shows how to install Antora's dependencies. Given that, it then restructures the headers in the remainder of the document to make the content make sense structurally.

💁‍♂️ this PR contains two distinct changes, _which it shouldn't_.

### CODE REVIEW

File: docs/build-the-docs.md [hunk 1]
```diff
@@ -74,7 +88,7 @@ DOCSEARCH_ENABLED=true DOCSEARCH_ENGINE=lunr antora --pull \
 
 - You can add the `--clean` option to clean the build directory of any leftover artifacts from the previous build, including PDF's.
 
-### Update The Generated Search Index
+#### Update The Generated Search Index
```
Comment (mmattel) [inline docs/build-the-docs.md]: imho we have to do something better / clarifying here as it is very confusing when you look downwards to `update-the-search-index.md`

File: docs/update-the-search-index.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Update the Search Index
+
+To update the search index (likely after a change to the documentation), you need to run the following command, adding the relevant values from [ownCloud’s Algolia account](https://www.algolia.com/apps/RQPV9Q61S4/api-keys/all).
+
+**Note:** if you do not have access to ownCloud’s Algolia account, please contact tboerger@owncloud.com.
+```
+docker run \
+    -v $(pwd)/docsearch-scraper:/opt/docsearch-scraper:ro \
```
Comment (mmattel) [inline docs/update-the-search-index.md]: Where does docker run? If locally, something needs to be added as installation prerequisite...

Comment (settermjd) [reply docs/update-the-search-index.md]: I've made an update which should solve this issue.

File: docs/update-the-search-index.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+```
+
+This command runs Algolia’s scrapes the documentation and builds a new search index based on the information found; a search index designed and optimized *specifically* for documentation sites.
+
+**Note:** To run this command, you will need [Docker](https://docs.docker.com/) installed.
+
+The configuration file for the script is stored in `algolia-config.json`, located in the root directory of ownCloud's docs repository.
+The configuration stores the documentation base URL, along with CSS selectors to help the scraper work with the site’s content structure.
+
+**Note:** Normally, manually updating is not required, as the CI pipeline manages this process, automatically.
```
Comment (mmattel) [inline docs/update-the-search-index.md]: Why not moving that note to the top of the page, would make more sense to me.

Comment (settermjd) [reply docs/update-the-search-index.md]: I'll have a re-read, but it sounds like a fair request.

File: docs/build-the-docs.md (hunk 1 above)
Comment (mmattel) [reply docs/build-the-docs.md]: I mean a better explanation why a search index needs updating

File: docs/build-the-docs.md [hunk 4]
```diff
@@ -1,16 +1,30 @@
...
 
 There are two ways to generate the documentation in HTML format:
 
 - Using ownCloud's custom Antora Docker Container
 - Running Antora from the Command-Line
 
-### Using the Docker Container
+#### Using the Docker Container
 
 To build the documentation using the Docker container, from the command line, in the root of the docs directory, run the following command:
```
Comment (mmattel) [inline docs/build-the-docs.md]: below in line 33/34 you find environment variables set for `lunr`. Is this still valid?

Comment (mmattel) [reply docs/build-the-docs.md]: Forget my comment, that is fixed in #475

File: docs/build-the-docs.md (hunk 1 above)
Comment (settermjd) [reply docs/build-the-docs.md]: This is fixed in #500. So long as it's merged after, then this will be resolved.

**Pull request merged**